### ISSUE TITLE
activation outputs do not match model summary outputs for stacked conv1d model

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#12301]
Hi there,

I'm trying to examine activation outputs in my model.  

Model is below.  

```python
maxlen=300
embedding_dim=100  

model = Sequential()
model.add(Conv1D(16, kernel_size  =2 , input_shape = (maxlen, embedding_dim),
                              activation='sigmoid', strides=1, name='conv1d_1'))
model.add(MaxPooling1D(3, name='max_pool'))
model.add(Conv1D(16, kernel_size  = 2 , 
                              activation='sigmoid', strides=1, name='conv1d_2'))
model.add(GlobalMaxPooling1D(name='global_max_pool'))
model.add(Dense(1,activation='sigmoid', name='output'))
model.compile(loss='binary_crossentropy', 
                           optimizer='rmsprop', metrics=['accuracy'])

print(model.summary())
```

```
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
conv1d_1 (Conv1D)            (None, 299, 16)           3216      
_________________________________________________________________
max_pool (MaxPooling1D)      (None, 99, 16)            0         
_________________________________________________________________
conv1d_2 (Conv1D)            (None, 98, 16)            528       
_________________________________________________________________
global_max_pool (GlobalMaxPo (None, 16)                0         
_________________________________________________________________
output (Dense)               (None, 1)                 17        
=================================================================
Total params: 3,761
Trainable params: 3,761
Non-trainable params: 0
_________________________________________________________________
None
```
I've verified the model weights match the number of parameters in the model_summary (16+2*100*16 for conv1d_1, 16 + 2*16*16, for conv1d_2, and 16 + 1 for the final dense layer).

When I run the code below on the activation outputs from the last layer (per Chollet's book, pp. 251-252) captured during calibration 

for layer in range(0,5):
    activation = np.load("%s/activations_at_epoch_%d_layer_%d.npy" %(log_dir, max_epoch-1, layer))
    print("activation index", layer, "shape", activation.shape)

I get:

activation index 0 shape (284, 299, 16)
activation index 1 shape (284, 99, 16)
activation index 2 shape (284, 16)
activation index 3 shape (284, 16)
activation index 4 shape (284, 1)

Note that the index 2 (conv1d) output is not of dimension (284,98,16), as one would expect from the model_summary output where n=284 is the size of the validation sample.

Additionally, the index 3 values do not lie within (-1,1) and I can reproduce the index 4 activations if I product sum the index 2 values with the parameter values (+ bias) and take the sigmoid.

Thanks for any help you can provide.
Sue

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
